pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/LinuxInfoUtils.java

```diff
@@ -72,6 +73,31 @@ public static boolean isCGroupEnabled() {
         }
     }
 
+    /**
+     *  Get total cpu count.
+     * @return Total cpu count
+     */
+    public static int getTotalCpuCount() {
+        int totalCpuCount=0;
+        try {
+            String[] ranges = readTrimStringFromFile(Paths.get(CGROUPS_CPU_CPUSET_CPUS)).split(",");
+            for (String range : ranges) {
+                if (!range.contains("-")) {
+                    totalCpuCount++;
+                } else {
+                    int dashIndex = range.indexOf('-');
+                    int left = Integer.valueOf(range.substring(0, dashIndex));
+                    int right = Integer.valueOf(range.substring(dashIndex + 1));
+                    totalCpuCount += right - left + 1;
+                }
+            }
+        } catch (IOException e) {
+            log.warn("[LinuxInfo] Failed to read CPU counts from cgroups", e);
+            // Fallback to availableProcessors
```

Comment:
How do we fallback to availableProcessor here? Dont we need to let the IOException throw here?

Reply:
you are right.

Reply:
Also please confirm if this can fallback to the next cpu limit getter logic,if it outputs 0 cpu count.